crates/red_knot_python_semantic/src/semantic_index/definition.rs

```diff
@@ -280,6 +299,9 @@ impl DefinitionNodeRef<'_> {
             Self::MatchPattern(MatchPatternDefinitionNodeRef { identifier, .. }) => {
                 identifier.into()
             }
+            Self::ExceptHandler(ExceptHandlerDefinitionNodeRef { symbol_name, .. }) => {
+                symbol_name.into()
```

Comment:
The idea here was that we don't really need the symbol name here at all, this could be the except handler node, or it could be the handled-exceptions expression, with no loss in correctness.

But, it does cost very little to use the symbol name here, and although it's not required in any way, it's arguably clearer for the definition node key to be the specific name/identifier node that is defined, when that works out. And, if we are going to pass through the handled-exceptions expression instead of the except handler node, then we do need `ExceptHandlerDefinitionNodeRef` either way (whether it contains the symbol name or not), because we can't just implement `From<ast::Expr> for DefinitionNodeRef` -- we wouldn't know what kind of expression we're dealing with. So I think what you currently have makes good sense.

Thanks for bearing with me! My main concern was to have as much parity as feasible between how we handle the different cases, but it turns out every case is a little different! This looks good.